lib/matplotlib/axes/_axes.py

```diff
@@ -2778,10 +2779,16 @@ def errorbar(self, x, y, yerr=None, xerr=None,
         .. plot:: mpl_examples/statistics/errorbar_demo.py
 
         """
-
-        if errorevery < 1:
+        
+        if not iterable(errorevery):
+            errorevery = (errorevery, 0)
+        try:
+            assert all(int(x)==x for x in errorevery)
```

Comment:
I while ago we cleaned all of the asserts out of the main code base and use a `if ... raise` pattern.